android/src/main/java/com/oney/WebRTCModule/GetUserMediaImpl.java

```diff
@@ -331,6 +123,83 @@ ReadableArray enumerateDevices() {
         return array;
     }
 
+    private ReactApplicationContext getReactApplicationContext() {
+        return reactContext;
+    }
+
+    MediaStreamTrack getTrack(String id) {
+        TrackPrivate private_ = tracks.get(id);
+
+        return private_ == null ? null : private_.track;
+    }
+
+    /**
+     * Implements {@code getUserMedia} without knowledge whether the necessary
+     * permissions have already been granted. If the necessary permissions have
+     * not been granted yet, they will be requested.
```

Comment:
It is not clear from reading the code what will happen if the permissions are requested, but denied by the user. Could you please explain ?

Reply:
Is my understanding correct that before native GUM is called there will first go a permissions request promise on the JS side (by calling native permissions along the way) ? So it would be kind of unexpected to have a permission request here ? How is this case handled ? Would it throw an exception ?

Reply:
The comment is wrong, we are not requesting any permissions.

Reply:
That will never happen because if a permission is not granted the constraints are removed. I think this requires an extra note there. I'll update the comment.

Reply:
Updated the docs, PTAL!